### ISSUE TITLE
lineinfile module : insertbefore should insert before first match of specified regular expression

### ISSUE THREAD

**Issue by dmcnaught** [ansible/ansible-modules-core#3648]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

lineinfile module
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### OS / ENVIRONMENT

Ubuntu
##### SUMMARY

insertbefore : If specified, the line will be inserted before the last match of specified regular expression
insertafter : If specified, the line will be inserted after the last match of specified regular expression

It seems to me that insertbefore should insert before the first match of the specified regexp. This would be consistent with the way that insertafter works (the reverse).

In our example we are adding lines to sshd_config that need to go before any ^Match blocks. If there are multiple match blocks then insertbefore won't create a valid sshd_config 
##### STEPS TO REPRODUCE

``````
lineinfile: dest=/etc/ssh/sshd_config regexp="^#?PermitUserEnvironment" line="PermitUserEnvironment no" insertbefore="^Match" state=present```
``````
##### EXPECTED RESULTS

If more than one Match block is in /etc/ssh/sshd_config, the line should be inserted before the first.
##### ACTUAL RESULTS

ssh throws an error:

```
/etc/ssh/sshd_config line 94: Directive 'PermitUserEnvironment' is not allowed within a Match block
```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dtolj** [comment]
Is there a possible workaround for this in order to insert before first match or replace first match.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by jctanner** [comment]
duplicate of https://github.com/ansible/ansible/issues/20582
